# Gene analysis

## Get genetics data

In [107]:
import abagen
atlas = abagen.fetch_desikan_killiany()

In [108]:
files = abagen.fetch_microarray(data_dir='/dagher/dagher9/filip/', donors='all', verbose=0)

In [109]:
expression_DKT = abagen.get_expression_data('/dagher/dagher11/filip/Downloads/OASIS-TRT-20_jointfusion_DKT31_CMA_labels_in_MNI152_2mm_v2.nii.gz',
                                            data_dir='dagher/dagher9/filip/',
                                            missing='interpolate')

/dagher/dagher12/filip/software/miniconda3py/envs/py3/lib/python3.10/site-packages/abagen/probes_.py:66: FutureWarning: In a future version, `df.iloc[:, i] = newvals` will attempt to set the values inplace instead of always setting a new array. To retain the old behavior, use either `df[df.columns[i]] = newvals` or, if columns are non-unique, `df.isetitem(i, newvals)`
  reannotated.loc[:, 'entrez_id'] = reannotated['entrez_id'].astype('int')
/dagher/dagher12/filip/software/miniconda3py/envs/py3/lib/python3.10/site-packages/abagen/probes_.py:757: FutureWarning: DataFrame.set_axis 'inplace' keyword is deprecated and will be removed in a future version. Use `obj = obj.set_axis(..., copy=False)` instead
  micro = micro.set_axis(symbols, axis=1, inplace=False)
/dagher/dagher12/filip/software/miniconda3py/envs/py3/lib/python3.10/site-packages/abagen/probes_.py:757: FutureWarning: DataFrame.set_axis 'inplace' keyword is deprecated and will be removed in a future version. Use `obj = obj.set_ax

In [121]:
pd.set_option('display.max_rows', 500)
expression_DKT

gene_symbol      A1BG  A1BG-AS1       A2M     A2ML1   A3GALT2    A4GALT  \
label                                                                     
4            0.467001  0.241979  0.562175  0.645510  0.635516  0.567473   
5            0.505112  0.408924  0.434041  0.539158  0.631222  0.580857   
6            0.532365  0.399017  0.334089  0.440267  0.456985  0.453232   
7            0.516784  0.228627  0.677912  0.572534  0.648151  0.625653   
10           0.406815  0.220683  0.669152  0.569524  0.561747  0.577350   
11           0.788399  0.425645  0.725498  0.587261  0.472289  0.579752   
12           0.752544  0.382341  0.621301  0.567901  0.393274  0.583941   
13           0.708084  0.278800  0.585201  0.592670  0.416663  0.467780   
14           0.417575  0.201129  0.643622  0.472946  0.673287  0.584370   
15           0.376253  0.227984  0.705517  0.583642  0.625109  0.598996   
16           0.386625  0.234290  0.678508  0.598164  0.605992  0.558846   
17           0.441299  0.345020  0.393948  0.603722  0.569828  0.601270   
18           0.707201  0.295608  0.564434  0.573620  0.609313  0.659144   
24           0.171364  0.214428  0.567837  0.579942  0.597236  0.444864   
26           0.673886  0.393526  0.483509  0.648742  0.369966  0.534722   
28           0.495163  0.233030  0.671308  0.514774  0.647662  0.578305   
30           0.640201  0.361113  0.308555  0.689112  0.364007  0.468533   
43           0.433116  0.386674  0.593463  0.684672  0.549104  0.450960   
44           0.458381  0.152612  0.674745  0.509638  0.716682  0.581188   
45           0.469298  0.308306  0.582911  0.592207  0.618100  0.603618   
46           0.516936  0.175758  0.748954  0.555027  0.634942  0.619300   
49           0.319822  0.218068  0.626842  0.541086  0.604148  0.508590   
50           0.626114  0.428771  0.610031  0.694526  0.526386  0.499759   
51           0.555250  0.345870  0.587268  0.555811  0.522481  0.558023   
52           0.541622  0.222298  0.664700  0.531354  0.499390  0.497971   
53           0.333827  0.188196  0.624980  0.555191  0.659486  0.620629   
54           0.446312  0.236849  0.685384  0.390871  0.517911  0.514046   
58           0.731545  0.462737  0.532069  0.543401  0.483830  0.459284   
60           0.426990  0.165986  0.684575  0.547852  0.681363  0.628283   
91           0.560225  0.369219  0.438901  0.612650  0.456746  0.474431   
92           0.593826  0.403293  0.555885  0.394942  0.524822  0.435325   
630          0.434172  0.392414  0.323338  0.473463  0.479428  0.474488   
631          0.595785  0.331273  0.493981  0.425279  0.546994  0.518202   
632          0.669578  0.164388  0.772536  0.562273  0.644232  0.640786   
1002         0.626187  0.604754  0.519426  0.582590  0.333440  0.403817   
1003         0.531893  0.638248  0.555223  0.425456  0.414944  0.529754   
1005         0.350470  0.581113  0.472526  0.414920  0.393016  0.446477   
1006         0.681031  0.511477  0.592793  0.523258  0.397512  0.452723   
1007         0.610562  0.670750  0.459640  0.421969  0.371181  0.434842   
1008         0.429848  0.628115  0.429528  0.479014  0.391321  0.470180   
1009         0.607035  0.713048  0.433578  0.497267  0.402986  0.382188   
1010         0.451552  0.624424  0.458144  0.490341  0.386507  0.451033   
1011         0.375598  0.572449  0.485349  0.474146  0.429051  0.487652   
1012         0.654495  0.661672  0.479067  0.414379  0.413246  0.508750   
1013         0.347033  0.552535  0.432802  0.515070  0.452532  0.437825   
1014         0.644752  0.666914  0.444388  0.427809  0.371000  0.492806   
1015         0.587474  0.689814  0.427173  0.458515  0.386718  0.404107   
1016         0.547740  0.551932  0.383242  0.450679  0.477171  0.415852   
1017         0.395478  0.623211  0.485428  0.561605  0.509749  0.452892   
1018         0.554185  0.664276  0.578748  0.446238  0.461963  0.450620   
1019         0.573936  0.642089  0.520185  0.482418  0.437010  0.551297   
1020         0.500243  0.

In [57]:
import numpy as np
np.savetxt('/dagher/dagher11/filip/Obesity_maps/data/expression_DKT.csv',expression_DKT, delimiter=',',fmt='%f')

In [65]:
np.savetxt('/dagher/dagher11/filip/Obesity_maps/data/abagenes.csv',list(expression_DKT.columns), delimiter=',',
          fmt='%s')

## DKT for BMI-CT map in HCP

order of parcels = 1002
        1003
        1005
        1006
        1007
        1008
        1009
        1010
        1011
        1012
        1013
        1014
        1015
        1016
        1017
        1018
        1019
        1020
        1021
        1022
        1023
        1024
        1025
        1026
        1027
        1028
        1029
        1030
        1031
        1034
        1035
        2002
        2003
        2005
        2006
        2007
        2008
        2009
        2010
        2011
        2012
        2013
        2014
        2015
        2016
        2017
        2018
        2019
        2020
        2021
        2022
        2023
        2024
        2025
        2026
        2027
        2028
        2029
        2030
        2031
        2034
        2035

structs=['caudalanteriorcingulate','caudalmiddlefrontal','cuneus','entorhinal','fusiform',
'inferiorparietal','inferiortemporal','isthmuscingulate','lateraloccipital','lateralorbitofrontal','lingual',
'medialorbitofrontal','middletemporal','parahippocampal','paracentral','parsopercularis',
'parsorbitalis','parstriangularis','pericalcarine','postcentral','posteriorcingulate','precentral',
'precuneus','rostralanteriorcingulate','rostralmiddlefrontal','superiorfrontal','superiorparietal',
'superiortemporal','supramarginal','transversetemporal','insula','caudalanteriorcingulate','caudalmiddlefrontal',
'cuneus','entorhinal','fusiform','inferiorparietal','inferiortemporal','isthmuscingulate','lateraloccipital',
'lateralorbitofrontal','lingual','medialorbitofrontal','middletemporal','parahippocampal',
'paracentral','parsopercularis','parsorbitalis','parstriangularis','pericalcarine','postcentral',
'posteriorcingulate','precentral','precuneus','rostralanteriorcingulate','rostralmiddlefrontal',
'superiorfrontal','superiorparietal','superiortemporal','supramarginal','transversetemporal','i

In [1]:
from netneurotools import datasets as nntdatasets
from neuromaps import datasets as nmdatasets
from neuromaps.parcellate import Parcellater
import pandas as pd
import neuromaps
from neuromaps import transforms
import numpy as np

dkt_gifti=neuromaps.images.annot_to_gifti('/dagher/dagher11/filip/Obesity_maps/data_old/DKT.annot')
parc = Parcellater(dkt_gifti, 'fsaverage', hemi='L')  # assuming mymap is in mni152 space
BMI=transforms.fsaverage_to_fsaverage('/dagher/dagher11/filip/Obesity_maps/data_old/BMI_lh.gii', '164k', hemi='L')
mymap_parc = parc.fit_transform(BMI, 'fsaverage', hemi='L')

np.savetxt('/dagher/dagher11/filip/Obesity_maps/data/BMI_CT_DKT_LH.csv', mymap_parc, delimiter=',', fmt='%f')

/dagher/dagher12/filip/software/miniconda3py/envs/py3/lib/python3.10/site-packages/nilearn/input_data/__init__.py:27: FutureWarning: The import path 'nilearn.input_data' is deprecated in version 0.9. Importing from 'nilearn.input_data' will be possible at least until release 0.13.0. Please import from 'nilearn.maskers' instead.
  warnings.warn(message, FutureWarning)


In [8]:
import pandas
import scipy

#DKT_np=pandas.DataFrame.to_numpy(mymap_parc)
scipy.io.savemat('/dagher/dagher11/filip/Obesity_maps/data/BMI_CT_DKT_RH.mat', 
                 {'BMI_CT_DKT_RH': mymap_parc})

In [5]:
from netneurotools import datasets as nntdatasets
from neuromaps import datasets as nmdatasets
from neuromaps.parcellate import Parcellater
import pandas as pd
from neuromaps import transforms

dkt_gifti=neuromaps.images.annot_to_gifti('/dagher/dagher11/filip/Obesity_maps/data_old/rh.DKT.annot')
parc = Parcellater(dkt_gifti, 'fsaverage', hemi='R')  # assuming mymap is in mni152 space
BMI=transforms.fsaverage_to_fsaverage('/dagher/dagher11/filip/Obesity_maps/data_old/BMI_rh.gii', '164k', hemi='R')
mymap_parc = parc.fit_transform(BMI, 'fsaverage', hemi='R')

np.savetxt('/dagher/dagher11/filip/Obesity_maps/data/BMI_CT_DKT_RH.csv', mymap_parc, delimiter=',', fmt='%f')

#### Sig maps

In [25]:
from netneurotools import datasets as nntdatasets
from neuromaps import datasets as nmdatasets
from neuromaps.parcellate import Parcellater
import pandas as pd
import neuromaps
from neuromaps import transforms
import numpy as np

dkt_gifti=neuromaps.images.annot_to_gifti('/dagher/dagher11/filip/Obesity_maps/data_old/DKT.annot')
parc = Parcellater(dkt_gifti, 'fsaverage', hemi='L')  # assuming mymap is in mni152 space
BMI=transforms.fsaverage_to_fsaverage('/dagher/dagher11/filip/Obesity_maps/data_old/sig_lh.gii', '164k', hemi='L')
mymap_parc = parc.fit_transform(BMI, 'fsaverage', hemi='L')

#mymap_parc=np.power(10,mymap_parc)

#np.savetxt('/dagher/dagher11/filip/Obesity_maps/tables/Sig_BMI_CT_DKT_LH.csv', mymap_parc, delimiter=',', fmt='%f')

In [29]:
#mymap_parc=np.power(mymap_parc,10)
mymap_parc**10

array([4.3905652e-01, 4.4095059e+03, 8.3849404e-04, 5.7600852e-04,
       4.9809128e-04, 1.8260664e-01, 8.0135623e-03, 4.0112304e-06,
       2.3478784e-03, 1.5383643e+02, 5.6391297e-21, 1.0202330e+01,
       3.4824245e-07, 4.0145865e-06, 6.1680969e-02, 8.4737064e-03,
       3.5292008e+00, 1.8038536e+01, 3.5476114e-06, 3.6251180e+00,
       3.1199444e-05, 1.6779697e-01, 1.4118021e+01, 1.9189417e+02,
       9.9593242e+03, 2.5389436e+02, 4.1450924e+01, 1.4822268e-11,
       7.1101612e-04, 2.3065292e-08, 1.9199719e-08], dtype=float32)

In [39]:
import nibabel as nb

In [42]:
nb.gifti.gifti.GiftiImage(dkt_gifti)

NotImplementedError: Header class requires a conversion from <class 'nibabel.filebasedimages.FileBasedHeader'> to <class 'tuple'>

In [41]:
(dkt_gifti)

(<nibabel.gifti.gifti.GiftiImage at 0x7f31a755c880>,)

In [5]:
from netneurotools import datasets as nntdatasets
from neuromaps import datasets as nmdatasets
from neuromaps.parcellate import Parcellater
import pandas as pd
from neuromaps import transforms

dkt_gifti=neuromaps.images.annot_to_gifti('/dagher/dagher11/filip/Obesity_maps/data_old/rh.DKT.annot')
parc = Parcellater(dkt_gifti, 'fsaverage', hemi='R')  # assuming mymap is in mni152 space
BMI=transforms.fsaverage_to_fsaverage('/dagher/dagher11/filip/Obesity_maps/data_old/sig_rh.gii', '164k', hemi='R')
mymap_parc = parc.fit_transform(BMI, 'fsaverage', hemi='R')

np.savetxt('/dagher/dagher11/filip/Obesity_maps/tables/Sig_BMI_CT_DKT_RH.csv', mymap_parc, delimiter=',', fmt='%f')

## DKT for BMI-CT map in HCP Aging

In [14]:
from netneurotools import datasets as nntdatasets
from neuromaps import datasets as nmdatasets
from neuromaps.parcellate import Parcellater
import pandas as pd
import neuromaps
from neuromaps import transforms
import numpy as np

dkt_gifti=neuromaps.images.annot_to_gifti('/dagher/dagher11/filip/Obesity_maps/data_old/DKT.annot')
parc = Parcellater(dkt_gifti, 'fsaverage', hemi='L')  # assuming mymap is in mni152 space
BMI=transforms.fsaverage_to_fsaverage('/dagher/dagher11/filip/Obesity_maps/data_old/BMI_HCPA_lh.gii', '164k', hemi='L')
mymap_parc = parc.fit_transform(BMI, 'fsaverage', hemi='L')

np.savetxt('/dagher/dagher11/filip/Obesity_maps/data/HCPA_CT_DKT_LH.csv', mymap_parc, delimiter=',', fmt='%f')

In [15]:
import pandas
import scipy

#DKT_np=pandas.DataFrame.to_numpy(mymap_parc)
scipy.io.savemat('/dagher/dagher11/filip/Obesity_maps/data/HCPA_CT_DKT_LH.mat', 
                 {'HCPA_CT_DKT_LH': mymap_parc})

In [16]:
from netneurotools import datasets as nntdatasets
from neuromaps import datasets as nmdatasets
from neuromaps.parcellate import Parcellater
import pandas as pd
from neuromaps import transforms

dkt_gifti=neuromaps.images.annot_to_gifti('/dagher/dagher11/filip/Obesity_maps/data_old/rh.DKT.annot')
parc = Parcellater(dkt_gifti, 'fsaverage', hemi='R')  # assuming mymap is in mni152 space
BMI=transforms.fsaverage_to_fsaverage('/dagher/dagher11/filip/Obesity_maps/data_old/BMI_HCPA_rh.gii', '164k', hemi='R')
mymap_parc = parc.fit_transform(BMI, 'fsaverage', hemi='R')

np.savetxt('/dagher/dagher11/filip/Obesity_maps/data/HCPA_CT_DKT_RH.csv', mymap_parc, delimiter=',', fmt='%f')

In [17]:
import pandas
import scipy

#DKT_np=pandas.DataFrame.to_numpy(mymap_parc)
scipy.io.savemat('/dagher/dagher11/filip/Obesity_maps/data/HCPA_CT_DKT_RH.mat', 
                 {'HCPA_CT_DKT_RH': mymap_parc})

## Use Cammoun parcellation for BMI data - DBM

In [53]:
from netneurotools import datasets as nntdatasets
from neuromaps import datasets as nmdatasets
from neuromaps.parcellate import Parcellater
import pandas as pd
from neuromaps import transforms
import neuromaps
import numpy as np



cammoun = nntdatasets.fetch_cammoun2012(version="MNI152NLin2009aSym")
info = pd.read_csv(cammoun['info'])
labels = info.query('scale=="scale125"')['label']  # or might be "labels"

parc = Parcellater(cammoun['scale125'], 'mni152')  # assuming mymap is in mni152 space
BMI = neuromaps.images.load_nifti('/dagher/dagher9/filip/DBM/HCP_DBM_Slvl/beta_0003.nii')

mymap_parc = parc.fit_transform(BMI, 'mni152')

np.savetxt('/dagher/dagher11/filip/Obesity_maps/data/BMI_DBM_Cammoun125_LH_RH.csv', mymap_parc[0], delimiter=',', fmt='%f')

In [40]:
info.to_csv('/dagher/dagher11/filip/Obesity_maps/data/Cammoun_labels.csv')

In [54]:
parc

## SA and CT

In [85]:
from netneurotools import datasets as nntdatasets
from neuromaps import datasets as nmdatasets
from neuromaps.parcellate import Parcellater
import pandas as pd
from neuromaps import transforms
import neuromaps
import numpy as np



cammoun = nntdatasets.fetch_cammoun2012(version="fsaverage")
#info = pd.read_csv(cammoun['info'])
#labels = info.query('scale=="scale125"')['label']  # or might be "labels"
cammoun_gifti=neuromaps.images.annot_to_gifti(cammoun['scale125'])

parc = Parcellater(cammoun_gifti, 'fsaverage')  # assuming mymap is in mni152 space

BMI=transforms.fsaverage_to_fsaverage('/dagher/dagher11/filip/Obesity_maps/data/BMI_rh.gii', '164k', hemi='L')
mymap_parc = parc.fit_transform(BMI, 'fsaverage', hemi='L')

np.savetxt('/dagher/dagher11/filip/Obesity_maps/data/BMI_thickness_Cammoun125_RH.csv', mymap_parc, delimiter=',', fmt='%f')

In [75]:
from nibabel import freesurfer

BMI_HCP=list(['/dagher/dagher11/filip/Obesity_maps/data/BMI_rh.gii'])

BMI_ctx=transforms.fsaverage_to_fsaverage(BMI_HCP, '164k', hemi='L')
BMI=BMI_ctx[0].agg_data()

Cammoun125=freesurfer.io.read_annot('/home/bic/fmorys/nnt-data/atl-cammoun2012/fsaverage/atl-Cammoun2012_space-fsaverage_res-125_hemi-R_deterministic.annot')[0]

BMI_Cammoun125=list()

for i in range(110):
    BMI_Cammoun125.append(np.average(BMI[Cammoun125==i]))
    
BMI_Cammoun125.pop(0)
BMI_Cammoun125.pop(3)

np.savetxt('/dagher/dagher11/filip/Obesity_maps/data/BMI_thickness_Cammoun125_RH.csv', BMI_Cammoun125, delimiter=',', fmt='%f')

In [1]:
from netneurotools import datasets as nntdatasets
from neuromaps import datasets as nmdatasets
from neuromaps.parcellate import Parcellater
import pandas as pd
from neuromaps import transforms
import neuromaps
import numpy as np



cammoun = nntdatasets.fetch_cammoun2012(version="fsaverage")
#info = pd.read_csv(cammoun['info'])
#labels = info.query('scale=="scale125"')['label']  # or might be "labels"
cammoun_gifti=neuromaps.images.annot_to_gifti(cammoun['scale125'])

parc = Parcellater(cammoun_gifti, 'fsaverage')  # assuming mymap is in mni152 space

BMI=transforms.fsaverage_to_fsaverage('/dagher/dagher11/filip/Obesity_maps/data/BMI_area_lh.gii', '164k', hemi='L')
mymap_parc = parc.fit_transform(BMI, 'fsaverage', hemi='L')

np.savetxt('/dagher/dagher11/filip/Obesity_maps/data/BMI_area_Cammoun125.csv', mymap_parc, delimiter=',', fmt='%f')

/dagher/dagher12/filip/software/miniconda3py/envs/py3/lib/python3.10/site-packages/nilearn/input_data/__init__.py:27: FutureWarning: The import path 'nilearn.input_data' is deprecated in version 0.9. Importing from 'nilearn.input_data' will be possible at least until release 0.13.0. Please import from 'nilearn.maskers' instead.
  warnings.warn(message, FutureWarning)


## Both hemispheres CT

In [2]:
from netneurotools import datasets as nntdatasets
from neuromaps import datasets as nmdatasets
from neuromaps.parcellate import Parcellater
import pandas as pd
from neuromaps import transforms
import neuromaps
import numpy as np



cammoun = nntdatasets.fetch_cammoun2012(version="fsaverage")
#info = pd.read_csv(cammoun['info'])
#labels = info.query('scale=="scale125"')['label']  # or might be "labels"
cammoun_gifti=neuromaps.images.annot_to_gifti(cammoun['scale125'])

parc = Parcellater(cammoun_gifti, 'fsaverage')  

BMI=transforms.fsaverage_to_fsaverage(['/dagher/dagher11/filip/Obesity_maps/data/BMI_lh.gii',
                                       '/dagher/dagher11/filip/Obesity_maps/data/BMI_rh.gii'], 
                                      '164k')
mymap_parc = parc.fit_transform(BMI, 'fsaverage')

np.savetxt('/dagher/dagher11/filip/Obesity_maps/data/BMI_thickness_Cammoun125_LHRH.csv', mymap_parc, delimiter=',', fmt='%f')

/dagher/dagher12/filip/software/miniconda3py/envs/py3/lib/python3.10/site-packages/nilearn/input_data/__init__.py:27: FutureWarning: The import path 'nilearn.input_data' is deprecated in version 0.9. Importing from 'nilearn.input_data' will be possible at least until release 0.13.0. Please import from 'nilearn.maskers' instead.
  warnings.warn(message, FutureWarning)


## Education regressor for sanity checks

In [89]:
from netneurotools import datasets as nntdatasets
from neuromaps import datasets as nmdatasets
from neuromaps.parcellate import Parcellater
import pandas as pd
from neuromaps import transforms
import neuromaps
import numpy as np



cammoun = nntdatasets.fetch_cammoun2012(version="fsaverage")
#info = pd.read_csv(cammoun['info'])
#labels = info.query('scale=="scale125"')['label']  # or might be "labels"
cammoun_gifti=neuromaps.images.annot_to_gifti(cammoun['scale125'])

parc = Parcellater(cammoun_gifti, 'fsaverage')  

Edu=transforms.fsaverage_to_fsaverage(['/dagher/dagher11/filip/Obesity_maps/data/edu_lh.gii'], 
                                      '164k',hemi="L")
mymap_parc = parc.fit_transform(Edu, 'fsaverage', hemi='L')

np.savetxt('/dagher/dagher11/filip/Obesity_maps/data/Edu_thickness_Cammoun125_LH.csv', mymap_parc, delimiter=',', fmt='%f')